src/main/java/at/hannibal2/skyhanni/features/gui/customscoreboard/CustomScoreboard.kt

```diff
@@ -0,0 +1,127 @@
+//
+// TODO LIST
+// V2 RELEASE
+//  - Soulflow API
+//  - Bank API (actually maybe not, I like the current design)
+//  - beacon power
+//  - skyblock level
+//  - more bg options (round, blurr, outline)
+//  - countdown events like fishing festival + fiesta when its not on tablist
+//  - CookieAPI https://discord.com/channels/997079228510117908/1162844830360146080/1195695210433351821
+//  - Rng meter display
+//  - shorten time till next mayor https://discord.com/channels/997079228510117908/1162844830360146080/1216440046320746596
+//  - option to hide coins earned
+//  - color options in the purse etc lines
+//  - choose the amount of decimal places in shorten nums
+//
+
+package at.hannibal2.skyhanni.features.gui.customscoreboard
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiPositionMovedEvent
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.HorizontalAlignment
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStringsAlignedWidth
+import at.hannibal2.skyhanni.utils.TabListData
+import net.minecraftforge.client.GuiIngameForge
+import net.minecraftforge.client.event.RenderGameOverlayEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+typealias ScoreboardElementType = Pair<String, HorizontalAlignment>
+
+class CustomScoreboard {
+
+    private var display = emptyList<ScoreboardElementType>()
+    private var cache = emptyList<ScoreboardElementType>()
+    private val guiName = "Custom Scoreboard"
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (display.isEmpty()) return
+
+        RenderBackground().renderBackground()
+
+        if (!TabListData.fullyLoaded && config.displayConfig.cacheScoreboardOnIslandSwitch && cache.isNotEmpty()) {
+            config.position.renderStringsAlignedWidth(cache, posLabel = guiName)
+        } else {
+            config.position.renderStringsAlignedWidth(display, posLabel = guiName)
+
+            if (cache != display) cache = display
+        }
+    }
+
+    @SubscribeEvent
+    fun onGuiPositionMoved(event: GuiPositionMovedEvent) {
+        if (event.guiName == guiName) {
+            val alignmentConfig = config.displayConfig.alignment
+            if (alignmentConfig.alignRight || alignmentConfig.alignCenterVertically) {
+                alignmentConfig.alignRight = false
+                alignmentConfig.alignCenterVertically = false
+                ChatUtils.chat("Disabled Custom Scoreboard auto-alignment.")
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!isEnabled()) return
+
+        // Creating the lines
+        if (event.isMod(5)) {
+            display = createLines()
```

Comment:
Set the cache here instead
```suggestion
            display = createLines()
            if(TabListData.fullyLoaded){
               cache = display.toList()
            }
```